src/shogun/clustering/KMeansMiniBatch.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2014 Parijat Mazumdar
+ * Written (W) 2014 Saurabh Mahindre
+ */
+
+#include <shogun/clustering/KMeansMiniBatch.h>
+#include <shogun/clustering/KMeansMiniBatchImpl.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/distance/Distance.h>
+#include <shogun/features/DenseFeatures.h>
+
+using namespace shogun;
+
+namespace shogun
+{
+
+CKMeansMiniBatch::CKMeansMiniBatch() : CKMeansBase()
+{
+	init();
+}
+
+CKMeansMiniBatch::CKMeansMiniBatch(int32_t k_, CDistance* d, bool use_kmpp) : CKMeansBase(k_, d, use_kmpp)
+{
+	init();
+}
+
+CKMeansMiniBatch::CKMeansMiniBatch(int32_t k_i, CDistance* d_i, SGMatrix<float64_t> centers_i) : CKMeansBase(k_i, d_i, centers_i)
+{
+	init();
+}
+
+CKMeansMiniBatch::~CKMeansMiniBatch()
+{
+}
+
+void CKMeansMiniBatch::set_mbKMeans_batch_size(int32_t b)
+{
+	REQUIRE(b>0, "Parameter bach size should be > 0");
```

Comment:
Please make sure your error messages are proper. This one has a typo and doesnt print the given number